src/Orleans/Telemetry/ITelemetryClient.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Orleans.Runtime
+{
+    /// <summary>Represents an object used for producing telemetry</summary>
+    public interface ITelemetryClient
```

Comment:
I'm not sure ITelemetryClient is a good name. Maybe ITelemetryProducer? Since this is a public interface. I'd like more people's opinion.
What do you think ? @sergeybykov @jason-bragg .

Reply:
ITelemetrySource or ITelemetryProducer?

Reply:
I don't have a strong opinion on this.  ITelemetryClient, ITelemetryLogger, ITelemetrySink, ... ?

Reply:
I considered both, but from the user's perspective, the source or the producer can be seen as a thing you subscribe to, not the client you use to publish telemetry yourself... I thought that client was less ambiguous from the user's perspective. Still open for debate, but wanted to share my reasoning after considering all of those.
As a 2 cent, client is also the name that AppInsights uses for their client api.

Reply:
I didn't like Client mainly because it kind of imply that there's TelemtryService, which is the AppInsight case, but not our case. In our case, this is just a abstraction for people to use to pipe telemetry to their chosen telemetry service. 

But I do agree your concern on 'producer'. So actually ITelemetrySink sounds better to me

Reply:
ITelemetrySource is good with me too.

Reply:
Guys, remember that it is a **very** temporary/transitory PR. I'm going to redo the telemetry APIs as discussed with @jdom before, right after this PR get merged so I can try have it merged in time for 2.0 release. So this interface will be dropped right away so I think the name shouldn't matter.

Reply:
@galvesribeiro still, anything that goes in needs to be in shippable state.
All: I'll rename to ITelemetryProducer. I don't think it's better than ITelemetryClient, but at least is not as ambiguous as Source. Sink would be the entire opposite... the sinks in a logging provider system are the log consumers

Reply:
@galvesribeiro hopefully this will be a transitory PR for 2.0. I'd like we have something better for telemetry in 2.0. But no commitment was made there ASAIK. That's why I insisted on the naming, since there's a good chance that this is going to be part of 2.0

Reply:
@xiazen yeah, I know there is no commitments on changing it from the team. I committed myself with @jdom and others that I will do it before 2.0, so unless people is using bleeding edge TP nugets, or building themselves from source, nobody will even notice it. I was waiting @jason-bragg PR with facets and this PR to be merged so I can work on it.